### ISSUE TITLE
Pull time is not relevant with proxy-cache

### ISSUE THREAD

**Issue by gaglimax** [goharbor/harbor#20844]
**Expected behavior and actual behavior:**
When I set up a proxy-cache to Dockerhub, sometimes, the artifact is in a folder (cf. ollama/ollama), sometimes, it's not (cf. library/alpine) :
![image](https://github.com/user-attachments/assets/5e9d6896-2b6b-4325-8d90-196163b9d0e6)
![image](https://github.com/user-attachments/assets/c74c3799-6c14-458f-a0ec-190e390e94a2)

When the artifact is in a folder, we can see that the pull time is not updated on the folder artifact, whereas it is updated on artifact under the folder : 
![image](https://github.com/user-attachments/assets/4a29ca26-a049-4988-adbf-4912ab2de80f)

This result in a not desired daily purge on artifact if we set up this retention rule : 
![image](https://github.com/user-attachments/assets/e20e4631-3b42-481a-93a4-3906a51e9e28)

**Steps to reproduce the problem:**
- Create a proxy-cache to Dockerhub.
- Pull an image to create a folder artifact (sometimes, a folder is created, sometimes not but I don't know why).
- Check the there is not pull time on the folder.
- Dry run the retention rule to see that the folder will be deleted.

**Versions:**
Please specify the versions of following systems.

- harbor version: 2.11.0

**Comment by stonezdj** [comment]
It is a bug, for the image index artifact type, the pull time is always zero in proxy cache project, and the tag retention rule can not get the pull time to evaluate the rule.

**Comment by gaglimax** [comment]
Are there any plans to fix this bug ?
The purpose of this proxy cache project is to avoid reaching the pull limit rate on Dockerhub. Because of this bug, we cannot prevent it.

**Comment by Jean-Daniel** [comment]
I agree, it makes the whole point of using a cache moot, as it is purged every time the retention policy runs.
